zone/client_process.cpp

```diff
@@ -545,21 +559,30 @@ bool Client::Process() {
 	}
 
 	if (client_state != CLIENT_LINKDEAD && !eqs->CheckState(ESTABLISHED)) {
-		OnDisconnect(true);
```

Comment:
It was to prevent bouncing to offline state before I could accurately set linkdead state. I'll have to go back over the commit history for it but the commit message should have more details. IIRC it was the only way I could accurately preserve an expedition leader that goes linkdead otherwise it would detect it as offline immediately and choose a new one